periodic sync upstream KF to midstream ODH

tarilabs opened the pull request:
- keep https://github.com/kubeflow/model-registry/pull/467
- adapt https://github.com/kubeflow/model-registry/pull/446
see below
- keep https://github.com/kubeflow/model-registry/pull/466
- keep https://github.com/kubeflow/model-registry/pull/451
- keep https://github.com/kubeflow/model-registry/pull/470
- keep https://github.com/kubeflow/model-registry/pull/469
- keep https://github.com/kubeflow/model-registry/pull/471
- keep https://github.com/kubeflow/model-registry/pull/472
- keep https://github.com/kubeflow/model-registry/pull/475
- 3c040aea22a8e0943a017f91030c126e19376099 adapt #446 to midstream ODH

merge label is applied.

superseeds https://github.com/opendatahub-io/model-registry/pull/133

@ederign @Griffin-Sullivan @alexcreasy can you kindly double-check 3c040aea22a8e0943a017f91030c126e19376099 as I'm sync manually for this round upstream to midstream.

isinyaaa reviewed (approved):
lgtm, thanks for the fix, Matteo :)

lampajr reviewed (approved):
lgtm

lucferbux committed:
Add archived model to mocks (#467)

Signed-off-by: lucferbux <lferrnan@redhat.com>

Griffin-Sullivan committed:
Add GitHub Action for building and pushing UI and BFF images (#446)

Signed-off-by: Griffin-Sullivan <gsulliva@redhat.com>

jenny-s51 committed:
Fix UI Bugs for Labels (#466)

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

fix chips

jenny-s51 committed:
Apply MUI theming to Register Model Form (#451)

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

fix tests

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

add fieldset to custom components

fix spacing issue with helper text

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

remove comment

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

fixes for text area

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

format

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

add model description fix

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

create reusable fieldset component

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

move fieldset to components

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

add prop for className

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

lint

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

fix import warnings

fix TS error with FormFieldset.tsx

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

fix import order errors

lucferbux committed:
Add mocked properties to models and versions (#470)

Signed-off-by: lucferbux <lferrnan@redhat.com>

alexcreasy committed:
Update README with troubleshooting instructions for macOS (#469)

Signed-off-by: Alex Creasy <alex@creasy.dev>

jenny-s51 committed:
Apply MUI theming to "Versions" and "Details" Screens (#471)

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

apply theming to model info screens

fix button color on hover

add comment

fix button color on hover

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

format

update to absolute path

dependabot[bot] committed:
build(deps-dev): bump typescript from 5.6.2 to 5.6.3 in /clients/ui/frontend (#472)

Bumps [typescript](https://github.com/microsoft/TypeScript) from 5.6.2 to 5.6.3.
- [Release notes](https://github.com/microsoft/TypeScript/releases)
- [Changelog](https://github.com/microsoft/TypeScript/blob/main/azure-pipelines.release.yml)
- [Commits](https://github.com/microsoft/TypeScript/compare/v5.6.2...v5.6.3)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump @mui/material from 6.1.2 to 6.1.3 in /clients/ui/frontend (#475)

Bumps [@mui/material](https://github.com/mui/material-ui/tree/HEAD/packages/mui-material) from 6.1.2 to 6.1.3.
- [Release notes](https://github.com/mui/material-ui/releases)
- [Changelog](https://github.com/mui/material-ui/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mui/material-ui/commits/v6.1.3/packages/mui-material)

---
updated-dependencies:
- dependency-name: "@mui/material"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tarilabs committed:
Merge sync remote-tracking branch 'upstream/main' into tarilabs-20241015-sync

tarilabs committed:
Adapt #446 for midstream ODH

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

tarilabs committed:
gha: delete build-and-push-ui-images.yml

the action is not needed in midstream
plan is to merge everything into one ODH dashboard image for midstream

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

tarilabs commented:
clarified with dashboard team offline
implemented with d574a27
